xdc-plugin/xdc.cc

```diff
@@ -96,17 +96,19 @@ struct GetPorts : public Pass {
 		}
 		// TODO handle more than one port
 		port_name = args.at(1);
-		char port[128];
+		std::string port_str;
+		char* port = const_cast<char*>(port_str.c_str());
```

Comment:
If you are using a const cast with the STL library you know you are doing it wrong.  There are actually two bugs here:
 - You should be using `.data()` rather than `.c_str()` to get access to a mutable string
 - When you call `.reserve`, the `std::string` may reallocate, causing the pointer in `port` to become freed memory.

Both errors can result in memory safety violations and must be fixed.

Reply:
I wouldn't call `.data()` mutable - http://www.cplusplus.com/reference/string/string/data/ and I am not sure there is any difference between `c_str()` and `data()` in the current standard.

Reply:
Ok, I state corrected. Seems that since C++17 there is in fact a non-const `.data()`

Reply:
However, we build the plugins with the same cxx_flags as yosys which currently uses c++11 so the non-const `.data()` is not there. Do we want to enable `-std=c++17` for the plugins?

Reply:
Regarding `.reserve` is [this](https://github.com/SymbiFlow/yosys-symbiflow-plugins/pull/4/commits/d474a7659304b2a7df88810cd6f50fe64726e769#diff-bf6d652c14576b97319c59ebd0933b30R99) acceptable?

Reply:
If we have to support C++11, then `&port[0]` is acceptable and legal.  Const casting an STL container is almost never legal.

Reply:
Fixed